website/src/content/docs/docs/libraries/http/files.md

```diff
@@ -0,0 +1,413 @@
+
+- `contents`: the contents of the file, which are the body of the request, response, or multipart payload. This location cannot be changed by subtypes of `Http.File`.
+- `contentType`: (optional) the media (MIME) type of the file, which is sent in the `Content-Type` header of the request, response, or multipart payload. This location cannot be changed by subtypes of `Http.File`.
+- `filename`: (optional) the name of the file, which is sent in the `filename` parameter of the `Content-Disposition` header of response and multipart payloads. By default, it cannot be sent in requests, as `Content-Disposition` is only valid for response and multipart payloads. This location _can_ be changed by subtypes of `Http.File` that apply HTTP metadata to the location. See [_Overriding the `filename` location_](#overriding-the-filename-location) below for more information.
+
+## Using `Http.File` in operations
+
+An operation payload (request, response, or multipart part) has a file body if:
+
+- The type of the body is _effectively_ a model that is or extends `Http.File` **AND**
+- there is no explicit declaration of a `Content-Type` header (see [_`File` with an explicit `Content-Type` header_ below for reasoning and more information](#file-with-an-explicit-content-type-header)).
+
+By "effectively a model that is or extends `Http.File`," we mean cases where an explicit body property is provided and its type is or extends `Http.File` as well as cases where `Http.File` is spread into a request or response payload and the only non-metadata properties in the payload are properties of `File` (see [_When a model is effectively a `File`_](#when-a-model-is-effectively-a-file) below for a more precise description with examples). The following sections contain examples of using `Http.File` in various contexts to define operations that have file bodies.
+
+### Downloading a file
+
+All of the following TypeSpec operation definitions have file bodies in the response:
+
+```typespec
+// The response is _exactly_ a File, so the response has a file body.
+op download(): File;
+
+// The response has an explicit body that is a File, so the response has a file body.
+op download(): {
+  @bodyRoot file: File;
+};
+
+// The response is _effectively_ a File (`File` is the only thing spread into it), so the response has a file body.
+op download(): {
+  ...File;
+};
+
+// The response has an explicit body that is _effectively_ a File, so the response has a file body.
+op download(): {
+  @bodyRoot file: {
+    ...File;
+  };
+};
+```
+
+### Uploading a file
+
+All of the following TypeSpec operation definitions have file bodies in the request:
+
+```typespec
+// The request has an explicit body that is _exactly_ a File, so the request has a file body.
+op upload(@bodyRoot file: File): void;
+
+// The request is _effectively_ a File (`File` is the only thing spread into it), so the request has a file body.
+op upload(...File): void;
+
```

Comment:
// File is intersected with other models containing only Http metadata, so the request has a file body.
alias FileRequest = {@header("x-request-id") requestId: string;} & File;
op download(...FileRequest): void;